enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,309 @@
+  which might turn off the console but also some other components).  This is what CLUSTER_PROFILES
+  sort of enable, but there seems to be reluctance to expand the cluster profile use case to include
+  these sorts of things.
+  b) enable/disable specific configurations such as "HA", where components could contribute multiple
+  deployment definitions for different configurations and then the installer/CVO would select the correct
+  one based on the chosen install configuration (HA vs single node) instead of having components read/reconcile
+  the infrastructure resource.
+
+2. How does the admin enable a component post-install if they change their mind about what components
+they want enabled?  Do we need/want to allow this?
+
+Turning on a component later is relatively easy (we expose a config resource for the CVO that defines
+the filter, we allow the user to remove items from the filter, the CVO will apply the previously
+filtered resources during the next reconciliation).
+
+Turning off a component later is more problematic because
+a) The CVO doesn't delete resources today, so that would be a new thing to teach it to do.
+b) Just deleting the resources for the component isn't sufficient, as the component also needs to clean
+itself up in case it created any additional resources on the cluster or contributed any configuration.
+
+
+3. What are the implications for upgrades if a future upgrade would add a component or resource which would
+have been filtered out during install time?  The install time choices need to be stored somewhere in
+the cluster(CVO config resource) and used to filter applied resources during upgrades also.  My understanding
+is today this is handled with CLUSTER_PROFILES and EXCLUDE_ANNOTATIONS by setting the env vars on the CVO
+pod, but if we want to allow the set to be changed (see (2), we need a more first class config
+resource that is admin editable)
+
+4. How prescriptive do we want to be about what can/can't be turned off?  Components need to opt into
+this by annotating their resources, so it's not completely arbitrary.
+
+5. What to do for components where disabling them has implications on other components or the way certain
+apis behave.  Example: disabling the internal registry changes the behavior of imagestreams
+(can't push to the imagestream anymore to push content to the internal registry) as well as the assumptions
+made by tools like new-app/new-build (create imagestreams that push to the internal registry).
+
+6. What to do(if anything) for components with interdependencies, to ensure a user doesn't break
+enabled components by disabling a dependency?  Options include:
+
+* Do nothing other than document dependencies so users know what not to turn off
+* Don't even annotate dependency resources for filtering, so if something is a dependency it cannot be turned off
+* Logic in the install or CVO that intelligently analyzes the filters the user has supplied and checks
```

Comment:
I agree that the installer should not be in the business of determining which resources to include or exclude based upon user-provided filters. As such, I like the annotation approach as laid out in this EP. I do think there is value in having the installer validate that the user has selected known addons to exclude. Since we will need to document for the end user which addons can be excluded, there is already a burden of maintaining the list of addons. It should not add much to that burden to also have the installer perform validation.

Reply:
for now i've added wording that the installer will not be responsible for doing this validation because there may be reasons we want to allow a user to specify addons that don't match anything currently.